## BUSINESS UNDERSTANDING
Airbus,wants to know Airplane crashes and fatalities occurences and causes in order to put in place prevention measures.As a data scientist I have been tasked to come up with various insights on the crashes hence helping Airbus to  put in place correct measures

## DATA UNDERSTANDING
The data was obtained from kaggle and it contains the following columns
- index,Date,Time,Location,Operator,Flight ,Route,Type,Registration,cn/In,Aboard,Fatalities,Ground,Summary
Date: Date of accident
Time: Local time, in 24 hr. in the format hh:mm

Location: Location of the accident

Operator: Airline or operator of the aircraft

Flight: Flight number assigned by the aircraft operator

Route: Complete or partial route flown prior to the accident

Type: Aircraft type

Registration: ICAO registration of the aircraft
cn/In: Construction or serial number / Line or fuselage number
Aboard: Total people aboard
Fatalities: Total people died

Ground: Total killed on the ground

Summary: Brief description of the accident and cause if known


In [184]:
#importing various libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
plt.style.use('ggplot')


In [185]:
##loading our dataframe
df=pd.read_csv('Airplane_Crashes_and_Fatalities_Since_1908.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5268 entries, 0 to 5267
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   index         5268 non-null   int64  
 1   Date          5268 non-null   object 
 2   Time          3049 non-null   object 
 3   Location      5248 non-null   object 
 4   Operator      5250 non-null   object 
 5   Flight #      1069 non-null   object 
 6   Route         3562 non-null   object 
 7   Type          5241 non-null   object 
 8   Registration  4933 non-null   object 
 9   cn/In         4040 non-null   object 
 10  Aboard        5246 non-null   float64
 11  Fatalities    5256 non-null   float64
 12  Ground        5246 non-null   float64
 13  Summary       4878 non-null   object 
dtypes: float64(3), int64(1), object(10)
memory usage: 576.3+ KB


The dataframe contains 14 columns, and  5267 entries

## DATA PREPARATION

In [186]:
##previewing our dataframe
df.head()

index        Date   Time                            Location  \
0      0  09/17/1908  17:18                 Fort Myer, Virginia   
1      1  07/12/1912  06:30             AtlantiCity, New Jersey   
2      2  08/06/1913    NaN  Victoria, British Columbia, Canada   
3      3  09/09/1913  18:30                  Over the North Sea   
4      4  10/17/1913  10:30          Near Johannisthal, Germany   

                 Operator Flight #          Route                    Type  \
0    Military - U.S. Army      NaN  Demonstration        Wright Flyer III   
1    Military - U.S. Navy      NaN    Test flight               Dirigible   
2                 Private        -            NaN        Curtiss seaplane   
3  Military - German Navy      NaN            NaN  Zeppelin L-1 (airship)   
4  Military - German Navy      NaN            NaN  Zeppelin L-2 (airship)   

  Registration cn/In  Aboard  Fatalities  Ground  \
0          NaN     1     2.0         1.0     0.0   
1          NaN   NaN     5.0         5.0     0.0   
2          NaN   NaN     1.0         1.0     0.0   
3          NaN   NaN    20.0        14.0     0.0   
4          NaN   NaN    30.0        30.0     0.0   

                                             Summary  
0  During a demonstration flight, a U.S. Army fly...  
1  First U.S. dirigible Akron exploded just offsh...  
2  The first fatal airplane accident in Canada oc...  
3  The airship flew into a thunderstorm and encou...  
4  Hydrogen gas which was being vented was sucked...

In [187]:
df.tail()

index        Date   Time  \
5263   5263  05/20/2009  06:30   
5264   5264  05/26/2009    NaN   
5265   5265  06/01/2009  00:15   
5266   5266  06/07/2009  08:30   
5267   5267  06/08/2009    NaN   

                                               Location  \
5263                             Near Madiun, Indonesia   
5264                  Near Isiro, DemocratiRepubliCongo   
5265  AtlantiOcean, 570 miles northeast of Natal, Br...   
5266       Near Port Hope Simpson, Newfoundland, Canada   
5267                  State of Arunachal Pradesh, India   

                             Operator Flight #  \
5263  Military - Indonesian Air Force      NaN   
5264                      Service Air      NaN   
5265                       Air France      447   
5266                       Strait Air      NaN   
5267      Military - Indian Air Force      NaN   

                                          Route  \
5263                           Jakarta - Maduin   
5264                               Goma - Isiro   
5265                     Rio de Janeiro - Paris   
5266  Lourdes de BlanSablon - Port Hope Simpson   
5267                         Mechuka for Jorhat   

                                  Type Registration cn/In  Aboard  Fatalities  \
5263           Lockheed C-130 Hercules       A-1325  1982   112.0        98.0   
5264                     Antonov An-26       9Q-CSA  5005     4.0         4.0   
5265                   Airbus A330-203       F-GZCP   660   228.0       228.0   
5266  Britten-Norman BN-2A-27 Islander      C-FJJR    424     1.0         1.0   
5267                     Antonov An-32          NaN   NaN    13.0        13.0   

      Ground                                            Summary  
5263     2.0  While on approach, the military transport cras...  
5264     NaN  The cargo plane crashed while on approach to I...  
5265     0.0  The Airbus went missing over the AtlantiOcean ...  
5266     0.0  The air ambulance crashed into hills while att...  
5267     0.0  The military transport went missing while en r...

In [188]:
df.describe().transpose()

count         mean          std  min      25%     50%      75%  \
index       5268.0  2633.500000  1520.884940  0.0  1316.75  2633.5  3950.25   
Aboard      5246.0    27.554518    43.076711  0.0     5.00    13.0    30.00   
Fatalities  5256.0    20.068303    33.199952  0.0     3.00     9.0    23.00   
Ground      5246.0     1.608845    53.987827  0.0     0.00     0.0     0.00   

               max  
index       5267.0  
Aboard       644.0  
Fatalities   583.0  
Ground      2750.0

In [189]:
df.dtypes

index             int64
Date             object
Time             object
Location         object
Operator         object
Flight #         object
Route            object
Type             object
Registration     object
cn/In            object
Aboard          float64
Fatalities      float64
Ground          float64
Summary          object
dtype: object

In [190]:
#checking for missing values
df.isna().sum()

index              0
Date               0
Time            2219
Location          20
Operator          18
Flight #        4199
Route           1706
Type              27
Registration     335
cn/In           1228
Aboard            22
Fatalities        12
Ground            22
Summary          390
dtype: int64

In [191]:
df.isna().sum()/len(df)*100

index            0.000000
Date             0.000000
Time            42.122248
Location         0.379651
Operator         0.341686
Flight #        79.707669
Route           32.384207
Type             0.512528
Registration     6.359150
cn/In           23.310554
Aboard           0.417616
Fatalities       0.227790
Ground           0.417616
Summary          7.403189
dtype: float64

Flight# has the highest percentage of missing valuees,so we will drop it together with the index column

In [192]:
#dropping columns
df.drop(['index','Flight #'],axis=1,inplace=True)

In [193]:
df.head()

Date   Time                            Location  \
0  09/17/1908  17:18                 Fort Myer, Virginia   
1  07/12/1912  06:30             AtlantiCity, New Jersey   
2  08/06/1913    NaN  Victoria, British Columbia, Canada   
3  09/09/1913  18:30                  Over the North Sea   
4  10/17/1913  10:30          Near Johannisthal, Germany   

                 Operator          Route                    Type Registration  \
0    Military - U.S. Army  Demonstration        Wright Flyer III          NaN   
1    Military - U.S. Navy    Test flight               Dirigible          NaN   
2                 Private            NaN        Curtiss seaplane          NaN   
3  Military - German Navy            NaN  Zeppelin L-1 (airship)          NaN   
4  Military - German Navy            NaN  Zeppelin L-2 (airship)          NaN   

  cn/In  Aboard  Fatalities  Ground  \
0     1     2.0         1.0     0.0   
1   NaN     5.0         5.0     0.0   
2   NaN     1.0         1.0     0.0   
3   NaN    20.0        14.0     0.0   
4   NaN    30.0        30.0     0.0   

                                             Summary  
0  During a demonstration flight, a U.S. Army fly...  
1  First U.S. dirigible Akron exploded just offsh...  
2  The first fatal airplane accident in Canada oc...  
3  The airship flew into a thunderstorm and encou...  
4  Hydrogen gas which was being vented was sucked...

In [194]:
#replacing missing values in objects with zero
df.fillna({'Time':0,'Location':'Unknown','Operator':'Unknown','Route':'Unknown',
            'Type':'Unknown','Registration':'Unknown','cn/In':0,'Summary':'Unknown'},
            inplace=True)

In [209]:
df.head()

AttributeError: 'NoneType' object has no attribute 'head'

In [206]:
#filling missing values of numerical columns with their median
df['Aboard'].fillna(df['Aboard'].median(),inplace=True )



TypeError: 'NoneType' object is not subscriptable

In [203]:
df


In [205]:
df['Fatalities'].fillna(df['Fatalities'].mean(),inplace=True)

TypeError: 'NoneType' object is not subscriptable

In [198]:
#df=df['Ground'].fillna(df['Ground'].mean(),inplace=True)